astropy/cosmology/parameter.py

```diff
@@ -210,8 +214,93 @@ def register(fvalidate):
+              "unit": self.unit,
+              "equivalencies": self.equivalencies,
+              "fmt": self.format_spec,
+              "doc": self.__doc__}
+        # validator is always turned into a function, but for ``repr`` it's nice
+        # to know if it was originally a string.
+        fvalidate = (self._validate_repr if self._validate_repr in self._registry_validators
+                     else self.fvalidate)
+        kw["fvalidate"] = fvalidate
+        return kw
+
+    def clone(self, **kw):
+        """Clone this `Parameter`, changing any constructor argument.
+
+        Parameters
+        ----------
+        **kw
+            Passed to constructor. The current values, eg. ``fvalidate`` are
+            used as the default values, so an empty ``**kw`` is an exact copy.
+
+        Examples
+        --------
+        >>> p = Parameter()
+        >>> p
+        Parameter(derived=False, unit=None, equivalencies=[],
+                  fvalidate='default', fmt='.3g', doc=None)
+
+        >>> p.clone(unit="km")
+        Parameter(derived=False, unit=Unit("km"), equivalencies=[],
+                  fvalidate='default', fmt='.3g', doc=None)
+        """
+        # start with defaults, update from kw.
+        kwargs = {**self._init_arguments, **kw}
+        # all initialization failures, like incorrect input are handled by init
+        cloned = type(self)(**kwargs)
+        # transfer over the __set_name__ stuff. If `clone` is used to make a
+        # new descriptor, __set_name__ will be called again, overwriting this.
+        cloned._attr_name = self._attr_name
+        cloned._attr_name_private = self._attr_name_private
+
+        return cloned
+
+    def __eq__(self, other):
+        """Check Parameter equality. Only equal to other Parameter objects.
+
+        Returns
+        -------
+        NotImplemented or True
+            `True` if equal, `NotImplemented` otherwise. This allows `other` to
+            be check for equality with ``other.__eq__``.
+
+        Examples
+        --------
+        >>> p1, p2 = Parameter(unit="km"), Parameter(unit="km")
+        >>> p1 == p2
+        True
+
+        >>> p3 = Parameter(unit="km / s")
+        >>> p3 == p1
+        False
+
+        >>> p1 != 2
+        True
+        """
+        if not isinstance(other, Parameter):
+            return NotImplemented
+        # check equality on all @property
+        for n in self._properties:  # pass to ``other.__eq__`` if fails check
+            if not hasattr(other, n) or (getattr(self, n) != getattr(other, n)):
+                return NotImplemented
+        return True  # everything passed!
+
     def __repr__(self):
-        return f"<Parameter {self._attr_name!r} at {hex(id(self))}>"
+        """String representation.
+
+        ``eval(repr())`` should work, depending if contents like ``fvalidate``
+        can be similarly round-tripped.
+        """
+        s = (f"Parameter(derived={self.derived!r}, "
```

Comment:
It seems you could use `_init_arguments` in here too, as in
```
s = "Parameter({})".format(", ".join([f"{key}={value}" for key, value in self._init_arguments.items()]))
```

Reply:
Great!
I've put that in, adapting it to force ``repr``

``s = "Parameter({})".format(", ".join([f"{k}={v!r}" for k, v in self._init_arguments.items()]))``